upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1424,16 +1405,39 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 			}
 			for _, additionalIP := range apiserverAdditionalIPs {
 				if cidr.Contains(net.ParseIP(additionalIP)) {
-					bootConfig.APIServer = additionalIP
+					bootConfig.APIServerIP = additionalIP
 					break
 				}
 			}
-			if bootConfig.APIServer != "" {
+			if bootConfig.APIServerIP != "" {
 				break
 			}
 		}
 	}
 
+	useConfigServer := apiModel.UseKopsControllerForNodeBootstrap(cluster) && (role != kops.InstanceGroupRoleMaster)
+	if useConfigServer {
+		host := "kops-controller.internal." + cluster.ObjectMeta.Name
+		if cluster.UsesNoneDNS() {
+			host = bootConfig.APIServerIP
```

Comment:
This is where I think it would be neat to have a custom dialer.  I'll see if I can find the code I was thinking of!